lib/vwo/core/variation_decider.rb

```diff
@@ -53,9 +54,56 @@ def initialize(settings_file, user_storage_service = nil)
       # @return[String,String]                       ({variation_id, variation_name}|Nil): Tuple of
       #                                              variation_id and variation_name if variation allotted, else nil
 
-      def get_variation(user_id, campaign, campaign_key, custom_variables = {})
+      def get_variation(user_id, campaign, api_name, campaign_key, custom_variables = {}, variation_targeting_variables = {})
         campaign_key ||= campaign['key']
 
+        return unless campaign
+
+        if campaign['isForcedVariationEnabled']
+          variation = evaluate_whitelisting(
+            user_id,
+            campaign,
+            api_name,
+            campaign_key,
+            variation_targeting_variables
+          )
+          if variation
+            status = StatusEnum::PASSED
+            variation_string = variation['name']
+          else
+            status = StatusEnum::FAILED
+            variation_string = ''
+          end
+
+          @logger.log(
+            LogLevelEnum::INFO,
+            format(
+              LogMessageEnum::InfoMessages::SEGMENTATION_STATUS,
```

Comment:
is this for passing whitelisting ? is log making sense?

Reply:
yes, logs are exact similar to node.js sdk.